### PR TITLE
[Triton] NUM_KSPLIT=1 for M<=32 preshuffled fp4 GEMM (N=8192, K=8192) (fixes #4867)

### LINKED ISSUE #4867: [Triton] gemm_afp4wfp4 preshuffle split-K OOB (GPU memory fault) on gfx950 for M=32/N=8192/K=8192 (regressed by #4630)
## Summary

On gfx950, the Triton `gemm_afp4wfp4_preshuffled_weight_scales` faults with a **GPU memory access fault -> SIGABRT** for `(M=32, N=8192, K=8192)`. `(M=64, N=4096, K=8192)` and the same shape at `M<32` pass. Regressed by **#4630**.

## Repro (aiter v0.1.20, gfx950 / MI355X, ROCm 7.2.3, torch 2.12)

The fault is shape/stride-driven (independent of the quant/shuffle values), so random `uint8` tensors of the expected shapes are enough:

```python
import torch
from aiter.ops.triton.gemm.basic.gemm_afp4wfp4 import gemm_afp4wfp4_preshuffled_weight_scales

torch.set_default_device("cuda")
M, N, K = 32, 8192, 8192
u8 = lambda *s: torch.randint(0, 256, s, dtype=torch.uint8)
x_fp4    = u8(M, K // 2)        # (32, 4096)    packed fp4 activations
w_fp4    = u8(N // 16, K * 8)   # (512, 65536)  preshuffled fp4 weights
x_scales = u8(M // 4, K)        # (8, 8192)     activation scales
w_scales = u8(N // 32, K)       # (256, 8192)   preshuffled weight scales
y = torch.empty(M, N, device="cuda", dtype=torch.bfloat16)
gemm_afp4wfp4_preshuffled_weight_scales(x_fp4, w_fp4, x_scales, w_scales, torch.bfloat16, y)
# -> Memory access fault by GPU node
```

Forcing `NUM_KSPLIT=1` on these exact shapes passes.

## Cause (bisected v0.1.19..v0.1.20)

First bad commit: **#4630** ("[Triton] unify the loading and add a UT for configs"). It corrects the `K` passed to `_get_config` for the preshuffle path, so the lookup now hits `configs/gemm/gfx950-GEMM-AFP4WFP4_PRESHUFFLED-N=8192-K=8192.json`, whose `M_LEQ_32` entry sets `NUM_KSPLIT=4` (every other M-bucket is 1). That exposes a latent split-K bug in `_gemm_afp4wfp4_preshuffle_kernel`:

- `get_splitk(K_elems=8192, BLOCK_SIZE_K=1024, NUM_KSPLIT=4)` -> `SPLITK_BLOCK_SIZE=4096`, i.e. `SPLITK_BLOCK_SIZE * NUM_KSPLIT = 16384 = 2 * K_elems` (overshoot).
- The A/B pointers use `SPLITK_BLOCK_SIZE // 2` (fp4 byte domain), but the split-K guard and the weight-scale offset (`offs_ks = pid_k * (SPLITK_BLOCK_SIZE // SCALE_GROUP_SIZE) * 32 + ...`) do not apply the same halving, so surplus split-K program ids read past A/B/scales -> OOB.

Isolation: `NUM_KSPLIT=1` runs fine (any `BLOCK_SIZE_K`); `NUM_KSPLIT=4` faults.

## Recommendation

1. Fix `get_splitk`'s 2x overshoot and make the split-K guard + scale offset K-domain consistent with the A/B `// 2` path.
2. Interim: set `NUM_KSPLIT=1` for the affected preshuffle configs. Perf note - `NUM_KSPLIT=1` is ~28-30 us flat at M=1/8/32 for this shape, so disabling the (currently broken) split-K shows no measured loss.

_Bisect and repro prepared with AI assistance._

### PR DESCRIPTION
**Pull request by Rohan138**
Fixes #4867.

The `M_LEQ_32` tuned entry for the `N=8192, K=8192` preshuffled fp4 GEMM selected `NUM_KSPLIT=4`. The preshuffled-scales kernel's split-K weight-scale load is not K-masked (and `get_splitk` overshoots), so `NUM_KSPLIT > 1` reads past the scale buffer and faults for small M. #4630 started selecting this config, turning the latent OOB into a GPU memory access fault.

Pin this shape to `NUM_KSPLIT=1` until the kernel's split-K scale handling is fixed. Other preshuffled configs still carry `NUM_KSPLIT>1`; this patches only the shape confirmed to crash.

**Repro** (gfx950): `M=32, N=8192, K=8192` preshuffled fp4 GEMM → GPU memory access fault before, clean after. Unblocks vLLM `test_rocm_mxfp4.py::test_aiter_fp4_gemm_preshuffled_tuned_shapes[shape1]`.

AI assistance was used; reviewed by the submitter.

### CODE REVIEW

**Review by azaidy — APPROVED** [review]
LGTM!

**Comment by Rohan138** [comment]
Fixed
